### PR TITLE
TE-1442: Add function to trigger on input field change in SearchForm

### PR DESCRIPTION
**Pull request by stephenkr**
[Related YouTrack issue](https://youtrack.lodgify.net/issue/TE-1442)

### What **one** thing does this PR do?
Implements a new prop in the `SearchForm` to trigger on input field change.

### Why this change?
This update is related to the implementation of the availability endpoint in the `SearchForm`. When an input field updates in the form, we want the availability shown to the user in the date picker to update.

### Any other notes
- Updated any component consuming the `SearchForm` to expose the input change prop

### CODE REVIEW

File: src/components/general-widgets/SearchBar/component.js [hunk 1]
```diff
@@ -22,7 +22,9 @@ export class Component extends PureComponent {
   state = {};
 
   persistInputChange = (name, value) => {
-    this.setState({ [name]: value });
+    this.setState({ [name]: value }, () => {
+      this.props.inputChange && this.props.inputChange(this.state);
```
Comment (marcdavi-es) [inline src/components/general-widgets/SearchBar/component.js]: In e.g. `InputController` we use `componentDidUpdate` for calling the `onChange` event after the state update. Let's be consistent with existing patterns..

File: src/components/general-widgets/SearchBar/component.js [hunk 2]
```diff
@@ -148,6 +152,8 @@ Component.propTypes = {
       ]),
     })
   ).isRequired,
+  /** A function to trigger when an input field changes */
+  inputChange: PropTypes.func,
```
Comment (marcdavi-es) [inline src/components/general-widgets/SearchBar/component.js]: Everywhere else we use `onChange*`.

File: src/components/general-widgets/SearchBar/component.js [hunk 3]
```diff
@@ -175,6 +186,8 @@ Component.propTypes = {
   modalSummaryElement: PropTypes.node,
   /** The element to be clicked to display the modal. */
   modalTrigger: PropTypes.node,
+  /** A function to trigger when an input field changes */
```
Comment (marcdavi-es) [inline src/components/general-widgets/SearchBar/component.js]: As before - let's be consistent with `InputController`. Use the prop documentation from there.

File: src/components/general-widgets/SearchBar/component.js [hunk 4]
```diff
@@ -21,6 +22,14 @@ import { getSearchBarModal } from './utils/getSearchBarModal';
 export class Component extends PureComponent {
   state = {};
 
+  componentDidUpdate(previousProps, previousState) {
+    const { onChangeInput } = this.props;
+
+    !isEqual(previousState, this.state) &&
+      onChangeInput &&
```
[the remark this answers is not in the document]
Comment (stephenkr) [reply src/components/general-widgets/SearchBar/component.js]: Yep good point.

File: src/components/general-widgets/SearchBar/component.js (hunk 3 above)
Comment (stephenkr) [reply src/components/general-widgets/SearchBar/component.js]: I'll update this. I'll try to be more consistent.

File: src/components/general-widgets/HomepageHero/component.js [hunk 5]
```diff
@@ -160,6 +164,8 @@ Component.propTypes = {
   ),
   /* The heading displayed in the search bar modal. */
   searchBarModalHeadingText: PropTypes.string,
+  /** The function to call when an input changes in the search bar */
```
Comment (marcdavi-es) [inline src/components/general-widgets/HomepageHero/component.js]: For this and all others, please use the existing props documentation from `InputController.props.onChange`

File: src/components/general-widgets/SearchBar/component.js (hunk 3 above)
Comment (marcdavi-es) [reply src/components/general-widgets/SearchBar/component.js]: do

File: src/components/general-widgets/HomepageHero/component.js (hunk 5 above)
Comment (stephenkr) [reply src/components/general-widgets/HomepageHero/component.js]: The change function will be passed the state, so I matched the props documentation for the `SearchBar.props.onSubmit`

**Pull request merged**